### DISCUSSION TITLE
DataLinq - Grundstücksabfrage

### DISCUSSION THREAD

**Discussion by Andreas-Lampert** [e-netze/datalinq-community#23] [Q&A]
Hallo Jürgen!

Ich würde gerne die Grundstücksabfrage im Imap auf DataLinq umbauen. 
Dazu habe ich drei Queries auf drei unterschiedliche Tabellen mit jeweils einer eigenen View gebaut.

- Query 1 - Grundstücksabfrage
- Query 2 - Eigentümer
- Query 3 - Nutzung

Alle drei Views liefern das gewünschte Ergebnis.
 
Der parametrisierte Aufruf schaut folgendermaßen aus:
.../grundstuecke-eigentuemer@select-grundstueck@grundstueck?KGNR=12345&EZ=12345&EL=12345. 
Mit der Kombination aus KGNR, EZ und EL bekomme ich das richtige Grundstück angezeigt. 

In der View "Grundstücksabfrage" würde ich jetzt gerne noch die beiden anderen Views einbinden. Das habe ich mit DLH.IncludeView probiert. 
Das funktioniert, wenn ich DLH.IncludeView(id: "grundstuecke-eigentuemer@select-eigentuemer@eigentuemer", filter: "EL=12345") schreibe, also den Filter-Wert absolut eintrage. Dann wird der Eigentümer der EL 12345 angezeigt. 

Ich muss aber den Filter dynamisch verwenden, d.h. der Wert von EL muss sich immer anpassen, je nachdem welches Grundstück ausgewählt wurde, damit der richtige Eigentümer angezeigt wird.

Gibt es eine Möglichkeit mit einem Aufruf gleichzeitig drei Views abzufragen und diese auf einer Seite einzubinden bzw. bei DLH.IncludeView den Filter-Parameter dynamisch zu verwenden?

Danke!

Lg, Andi

**Comment by fanie-mak** [comment]
Hallo Andi,

du übergibst ja bei deinem Aufruf die Parameter KGNR, EZ, EL... Diese übergibst du ja aus der Basisquery. Das passiert ja dann mit @record["KGNR"] oder so. Diese Werte kannst du auch bei IncludeView übergeben: 

@IntegratorHelper.IncludeView("ENDPOINT@QUERY@VIEW?KGNR="+@record["KGNR"]+"&EZ="+@record["EZ"]+"&EL="+@record["EL"], true)

Dann solltest du auch mehrere Views in einem Basisview öffnen können.

LG Steffi

**Comment by Andreas-Lampert** [comment]
Vielen Dank Steffi!! Ich hab bei meinem Probieren dauernd das + vergessen gehabt und deshalb einen Error bekommen. 

Lg, Andi
